coldfront/core/project/management/commands/add_service_units_to_project.py

```diff
@@ -0,0 +1,115 @@
+import logging
+from decimal import Decimal
+
+from django.core.management import BaseCommand, CommandError
+
+from coldfront.core.project.models import Project
+from coldfront.core.statistics.models import ProjectTransaction
+from coldfront.core.statistics.models import ProjectUserTransaction
+from coldfront.core.utils.common import utc_now_offset_aware
+from coldfront.api.statistics.utils import get_accounting_allocation_objects
+from coldfront.api.statistics.utils import set_project_allocation_value
+from coldfront.api.statistics.utils import set_project_user_allocation_value
+from coldfront.core.allocation.models import AllocationAttributeType, Allocation
+
+
+class Command(BaseCommand):
+    help = 'Command to add SUs to a given project.'
+    logger = logging.getLogger(__name__)
+
+    def add_arguments(self, parser):
+        parser.add_argument('--project_name',
+                            help='Name of project to add SUs to.',
+                            type=str,
+                            required=True)
+        parser.add_argument('--amount',
+                            help='Number of SUs to add to a given project.',
+                            type=int,
+                            required=True)
+        parser.add_argument('--reason',
+                            help='User given reason for adding SUs.',
+                            type=str,
+                            required=True)
+        parser.add_argument('--dry_run',
+                            help='Display updates without performing them.',
+                            action='store_true')
+
+    def handle(self, *args, **options):
+        """ Set user password for all users in test database """
+
+        project = Project.objects.get(name=options.get('project_name'))
```

Comment:
We should perform some validation on these inputs and inform the user about failing checks:
- If no corresponding `Project` exists,
- If the amount being added is not between `settings.ALLOCATION_MIN` and `settings.ALLOCATION_MAX`,
- If the new total after the addition is not in the same range, or
- If the reason isn't sufficiently long (let's say 20 characters).

Reply:
Sorry, the amount to be added should be allowed to be negative--but the new total should definitely be non-negative.